src/main/java/core/Winners.java

```diff
@@ -0,0 +1,71 @@
+package core;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class Winners {
+    private final List<Car> cars;
+
+    private Winners(List<Car> cars) {
+        this.cars = cars;
+    }
+
+    public static Winners decideWinners(Cars cars) {
+        CurrentWinners currentWinners = new CurrentWinners(new ArrayList<>());
+        for (Car car : cars.getCars()) {
+            decideCurrentWinners(currentWinners, car);
+        }
+        return new Winners(currentWinners.getCars());
+    }
+
+    private static void decideCurrentWinners(CurrentWinners currentWinners, Car car) {
+        if (currentWinners.isBehind(car)) {
+            currentWinners.changeWinner(car);
+            return;
+        }
+        if (currentWinners.isAtSamePosition(car)) {
+            currentWinners.addWinner(car);
+        }
+    }
```

Comment:
```suggestion
    CurrentWinners currentWinners;

    private Winners(CurrentWinners cars) {
        this.currentWinners = cars;
    }

    public static Winners of(Cars cars) {
        CurrentWinners currentWinners = new CurrentWinners(new ArrayList<>());
        for (Car car : cars.getCars()) {
            currentWinners.compare(car);
        }
        return new Winners(currentWinners);
    }
```
- CurrentWinners 객체를 적극적으로 활용해보는건 어떨까요?